### DISCUSSION TITLE
How to set always atribute for Custom Field?

### DISCUSSION THREAD

**Discussion by PayteR** [rebing/graphql-laravel#907] [Q&A] [answered]
Hi, how could i set always atribute for custom field? Now i have it like this
```
public function fields(): array
{
      return [
          'id' => [
              'type' => Type::nonNull(Type::int()),
              'description' => '',
          ],
          'custom' => CustomField:class
      ];
}
```
But my custom column is not always present in selected fields, so i want to set 
```
public function fields(): array
{
      return [
          'id' => [
              'type' => Type::nonNull(Type::int()),
              'description' => '',
          ],
          'custom' => [
              'type' => CustomField:class,
              'always' => ['custom_column']
          ]
      ];
}
```
But it gives me back error ` "Call to undefined function App\\GraphQL\\CustomField()"`. Any ideas how to achieve this? Thx

**Comment by mfn** [comment]
What is `CustomField` for a class definition, specifically?

I my first idea would be that the approach is wrong, do you have any reference how you got there?

Usually a type is either a native one like `Type::int()` or you reference you custom type via `GraphQL::type('<name')`, like the samples show in the README.

**Comment by PayteR** [comment, marked as answer]
No, i asking about how to retreive always column (field in this context) from database so i can use it in custom field https://github.com/rebing/graphql-laravel#custom-field . I think i got it, it's possible to set always attribute in CustomField

```
class CustomField extends Field
{
    protected $attributes = [
        'name' => 'CustomField',
        'description' => '',
        'always' => ['custom_column'],
    ];
```

Closing, thx anyway
